src/CMakeLists.txt

```
@@ -134,12 +134,23 @@ if(SECP256K1_BUILD_BENCHMARK)
 endif()
 
 if(SECP256K1_BUILD_TESTS)
+  include(CheckIncludeFiles)
+  check_include_files("sys/types.h" HAVE_SYS_TYPES_H)
+  check_include_files("sys/wait.h" HAVE_SYS_WAIT_H)
+  check_include_files("unistd.h" HAVE_UNISTD_H)
```

The [`CheckIncludeFile`](https://cmake.org/cmake/help/latest/module/CheckIncludeFile.html) module (in singular) appears more suitable:
```suggestion
  include(CheckIncludeFile)
  check_include_file(sys/types.h HAVE_SYS_TYPES_H)
  check_include_file(sys/wait.h HAVE_SYS_WAIT_H)
  check_include_file(unistd.h HAVE_UNISTD_H)
```

---

> Did you mark this "resolved" unintentionally?

I pushed hebasto's initial `CheckIncludeFile` suggestion after discussing the `CheckSymbolExists` one via DM, and forgot to update this thread (sorry for that). It just seemed to be an overkill for a feature we might introduce in a follow-up. Adding Windows support is not hard, it is just a bit more code.

---

> Adding Windows support is not hard, it is just a bit more code.

My apologies if I was misunderstood. I mentioned cross-compiling for Windows only as an example, not as the basis of my [comment](https://github.com/bitcoin-core/secp256k1/pull/1734#discussion_r2381543998). I do believe the build system should check the actual functionality we intend to use, which is the functions, rather than just the headers.

Here are a couple of examples for the `fork()` function from Bitcoin Core:
- in [Autotools](https://github.com/bitcoin/bitcoin/blob/a0b5730f858cf3d36d35105af473f7d0c83820e5/configure.ac#L909):
```autoconf
AC_CHECK_DECLS([fork])
```
- in [CMake](https://github.com/bitcoin/bitcoin/blob/25212dfdb4cd7291392b6a94130f658c5bfa0a48/cmake/introspection.cmake#L10):
```cmake
check_cxx_symbol_exists(fork "unistd.h" HAVE_DECL_FORK)
```

I'm fine with leaving this improvement for a follow-up PR.